openfermioncirq/trotter/algorithms/low_rank.py

```diff
@@ -259,29 +254,25 @@ def trotter_step(
                for j in range(n_qubits))
 
         # Simulate each singular vector of the two-body terms.
-        prior_basis_matrix = numpy.identity(n_qubits, complex)
+        prior_basis_matrix = numpy.identity(n_qubits)
         for j in range(len(self.eigenvalues)):
 
-            # Get the basis change matrix and its inverse.
+            # Get the two-body coefficients and basis change matrix.
             two_body_coefficients = self.scaled_density_density_matrices[j]
             basis_change_matrix = self.basis_change_matrices[j]
+            is_unitary = cirq.is_unitary(basis_change_matrix)
 
             # If the basis change matrix is unitary, merge rotations.
             # Otherwise, you must simulate two basis rotations.
-            is_unitary = cirq.is_unitary(basis_change_matrix)
             if is_unitary:
-                inverse_basis_matrix = numpy.conjugate(
-                    numpy.transpose(basis_change_matrix))
+                inverse_basis_matrix = basis_change_matrix.T.conj()
                 merged_basis_matrix = numpy.dot(prior_basis_matrix,
                                                 inverse_basis_matrix)
                 yield bogoliubov_transform(qubits, merged_basis_matrix)
             else:
-                # TODO add LiH test to cover this.
-                # coverage: ignore
-                if j:
-                    yield bogoliubov_transform(qubits, prior_basis_matrix)
+                yield bogoliubov_transform(qubits, prior_basis_matrix)
                 yield cirq.inverse(
```

Comment:
might as well be consistent with using either .T.conj() or cirq.inverse(), but not both, now that things are promised to be unitary.